PFW-1271 PF-build v2.0.0

3d-gussner opened the pull request:
PF-build.sh version 2.0.0

Added support for [MK404](https://github.com/vintagepc/MK404) see also [MK404 wiki](https://github.com/vintagepc/MK404/wiki)
- MK404-build.sh is only supported on LINUX at this moment
  - Checks for dependencies to build the MK404 SIM
  - Compiles the MK404 and some default SD cards.

Added ELF files being saved next to the hex files for debugging with FW3.10.1 and newer

PF-build.sh got few new arguments
- to start automatically the MK404 SIM with compiled Prusa-Firmware
  - `-m` 1 = Printer, 2 = Printer+MMU2
  -  `-g` 1 = lite, 2 = fancy graphics, 3 = lite with Quad_HR and 4=fancy with Quad_HR
  - `-x` 8/64 ram size for the virtual atmel404 in MK404
  - `-y` 256/384/512/1024/32M flash size for the virtual atmel404 in MK404
- Verbose mode
  - `-j` 1 = verbose output

Depends on https://github.com/prusa3d/Prusa-Firmware/pull/3093

ToDo before final PR:
- [x] Set MK404 `--debugcore` and `--bootloader-file` depending when extended RAM and FLASH size.

- [x] Remove MK404 copy of lang file as it has been fixed in MK404

3d-gussner committed:
Add MK404 support
Indentations

3d-gussner committed:
Add MK404 support
Indentations

3d-gussner committed:
Add force recompile
Added SD card builds

3d-gussner committed:
Add force recompile
Added SD card builds

3d-gussner committed:
Add copy of MK3 and MK3S `lang.bin` files to MK404 `*_xflash.bin`
Minor changes:
- Indentations
- Documentation

3d-gussner committed:
Add copy of MK3 and MK3S `lang.bin` files to MK404 `*_xflash.bin`
Minor changes:
- Indentations
- Documentation

3d-gussner committed:
Fix few issues
Create MMU2 SDcards
MMU2 only for MK3/S

3d-gussner committed:
Fix few issues
Create MMU2 SDcards
MMU2 only for MK3/S

3d-gussner committed:
Merge branch 'MK3_MK404' of https://github.com/3d-gussner/Prusa-Firmware into MK3_MK404

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into MK3_MK404

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into MK3_MK404

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into MK3_MK404

Fix merge

3d-gussner committed:
fix merge issue 2

3d-gussner committed:
Hack to get more flash and ram for MK404 debug

3d-gussner committed:
Improve MK404-build.sh check, update

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3_3.10.0' into MK3_MK404

3d-gussner committed:
Merge branch 'PFW-1189' into MK3_MK404

3d-gussner committed:
Save changes

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3_3.10.0' into MK3_MK404

3d-gussner committed:
Indentations

3d-gussner committed:
Add EN_ONLY support for MK404

3d-gussner committed:
Comment out "manual" xflash update as MK404 can do it now
directly from the hex file.

3d-gussner committed:
Update documentation

3d-gussner committed:
Add MK404 'atmega404' and 'atmega404_no_bootloder' option

3d-gussner committed:
Indentations

3d-gussner committed:
Merge branch 'MK3_more_flash_more_ram' into MK3_MK404

Add atmega404

3d-gussner committed:
Update MK404 part to '--bootloader-file ""'
Some cosmetics

3d-gussner committed:
Merge branch 'MK3_MK404' into PFW-1271_PF-buildv20

3d-gussner committed:
Update PF-build.sh to work after @DRracer Remove FW version parsing PR

3d-gussner committed:
Save ELF files for FW3.10.1 debugging PRs
Add verbose_IDE to output more information during build

3d-gussner committed:
Change version to v2.0.0

vintagepc commented:
I approve but perhaps I'm slightly biased here :rofl:

3d-gussner committed:
Use atmega404 if extanded RAM or FLASH size are chosen
Remove MK404 copy of lang file as it has been fixed in MK404

3d-gussner committed:
Documentation and version number
Added some arguments and checks

vintagepc commented on MK404-build.sh:
```diff
@@ -221,3 +312,115 @@ if [[ ! -f "$MK404_BUILD_PATH/Prusa_MK3S_SDcard.bin" || "$force_flag" == "1" ]];
+# Run MK404 with grafics
+    if [ ! -z "$graphics_flag" ]; then
+        if [ ! -z "$MK404_options" ]; then
+            MK404_options="${MK404_options} --colour-extrusion --extrusion Quad_HR -g "
```
Not sure if we want to make Quad_HR the default; perhaps better as an option too since the HRs can put a pretty hefty toll on the system for larger test prints.

3d-gussner committed:
Default extrusion graphics to line. Thanks @vintagepc point it out
Added -g 3 and 4 for more details extrusion lines
Check for updates is default. Fix update if internet connection is lost.

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into PFW-1271_PF-buildv20

3d-gussner committed:
Pf-build.sh
Change atmega404 board flash argument to y
Use newer version of MK404-build.sh instead start the program

MK404-build.sh
Change board_flash argument to 'y' and firmware_version to 'f

3d-gussner committed:
comment out debug echos

3d-gussner committed:
Defined OUTPUT_FILENAME in one location
Added _RAM-<value> and _FLASH-<value> as OUTPUT_FILE_SUFFIX if someone builds an atmega404 firmware
Added some logic if IDE 1.8.13 is used
Added some logic if Arduino_boards 1.0.4 is used
Fixed typo

3d-gussner committed:
Change atmega404 board flash argument to y
Use newer version of MK404-build.sh instead start the program
Added _RAM-<value> and _FLASH-<value> as OUTPUT_FILE_SUFFIX if someone builds an atmega404 firmware
Added some logic if IDE 1.8.13 is used
Added some logic if Arduino_boards 1.0.4 is used
Fixed typo
Restore original `Configuration.h` and `config.h` in case of cannceled script or failed compiling during next start of this script.
use function
Improve MK404 usage

3d-gussner committed:
Fix MK404 user interaction not to show if compiling 'All' variants

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into PFW-1271_PF-buildv20

merge latest changes and fix isses

3d-gussner committed:
MK404 is only supported on Linux at this moment.

3d-gussner committed:
Fix fresh build

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into PFW-1271_PF-buildv20

3d-gussner committed:
Add/fix few functions
Check MK404 for latest stable release instead of dev branch

3d-gussner committed:
Switch to arduino_boards v1.0.4

DRracer commented on MK404-build.sh:
```diff
@@ -0,0 +1,546 @@
+{
+packages=(
+"libelf-dev"
+"gcc-7"
```
@vintagepc do we really need to stick with gcc-7? At least gcc-10 works well

vintagepc replied:
If it compiles without issue there's no reason we have to stick to 7.5; it's just what my system and the build runners use. I can upgrade my system but I remember having to backtrack on the build runners because it was griping about something at the time. 

Nothing to it other than being a known constant for consistency reasons :)

DRracer committed:
Merge branch 'MK3_3.10.1' into PFW-1271_PF-buildv20
